components/hab/src/command/studio/enter.rs

```diff
@@ -114,6 +114,7 @@ mod inner {
                     let ident = PackageIdent::from_str(&format!("{}/{}",
                                                                 super::STUDIO_PACKAGE_IDENT,
                                                                 version[0]))?;
+                    let command = exec::command_from_min_pkg(ui, super::STUDIO_CMD, &ident)?;
                     // This is a duplicate of the code in `hab pkg exec` and
                     // should be refactored as part of or after:
                     // https://github.com/habitat-sh/habitat/issues/6633
```

Comment:
Is it maybe time for this refactoring?

Reply:
Yes :joy_cat:    I started down this path a couple weeks ago, but had to set it aside due to time constraints.  I've updated my notes on the tracking issue #6634 

In addition to the above refactor, I think we'd possibly want to pull the "ensure package is installed" code out of `command_from_min_pkg` so that we can isolate the responsibilities of that code. That function appears to only be used in a few places through the code base, but it is used by the launcher and supervisor, in addition to studio enter.

Reply:
> I think we'd possibly want to pull the "ensure package is installed" code out of `command_from_min_pkg`

Strong agree.

Replacing the `PackageInstall::load` on L121 with something like `::install_and_load` makes this more intuitive, I think. In fact, how often do we really want `PackageInstall::load` to fail if the package isn't installed versus trying the install? At a quick glance, it appears to be nearly always called with `?`. This might be the time to do a little audit of the callers and see if we have more latent bugs which are similar to this.

Reply:
I was thinking, based on how `PackageIdent` and `PackageInstall` are used here, that it would make sense to have method on `PackageIdent` that  returns a `PackageInstall` and handles the installation if needed.  

Perhaps there's a more Rusty way to handle it, but given that `PackageInstall::load` takes a PackageIdent, we probably have this pattern throughout the code and it, maybe naively, would make sense to have that behavior baked into the transformation of ident into install.

Reply:
If what we have is a `PackageIdent`, and want is a `PackageInstall`, the most rustic approaches to me seem to be adding `PackageInstall::from_ident() -> Self` or `PackageIdent::install(&self) -> PackageInstall`. I like the way the latter reads, but since `PackageInstall` has non-`pub` fields, you'll probably need to call on `PackageInstall` to create it for you. In that case implementing `PackageIdent::install` by having it call `PackageInstall::from_ident` might be appropriate.

But now I see that we also have an `InstallTask` type, which already has a `with_ident` method that returns a `Result<PackageInstall>`, so perhaps that's what we should be using. It's currently private, but it probably makes sense to expose at least part of that interface so we have a cleaner way to go from `PackageIdent` → `PackageInstall`. The complication is that [a number of other inputs are necessary](https://github.com/habitat-sh/habitat/blob/master/components/common/src/command/package/install.rs#L419-L427).